packages/grafana-ui/src/components/Table/Table.tsx

```diff
@@ -22,109 +31,142 @@ export interface Props {
+      switch (action.type) {
+        case 'columnDoneResizing':
+          if (props.onColumnResize) {
+            const info = (newState.columnResizing.headerIdWidths as any)[0];
+            const columnIdString = info[0];
+            const fieldIndex = parseInt(columnIdString, 10);
+            const width = Math.round(newState.columnResizing.columnWidths[columnIdString] as number);
+            props.onColumnResize(fieldIndex, width);
           }
-          return minWidth;
-        }, columnMinWidth),
-      }),
-      [columnMinWidth, memoizedColumns]
-    );
-
-    const options: any = useMemo(
-      () => ({
-        columns: memoizedColumns,
-        data: memoizedData,
-        disableResizing: !resizable,
-        defaultColumn,
-      }),
-      [memoizedColumns, memoizedData, resizable, defaultColumn]
-    );
-
-    const { getTableProps, headerGroups, rows, prepareRow, totalColumnsWidth } = useTable(
-      options,
-      useBlockLayout,
-      useResizeColumns,
-      useSortBy
-    );
-
-    const RenderRow = React.useCallback(
-      ({ index, style }) => {
-        const row = rows[index];
-        prepareRow(row);
-        return (
-          <div {...row.getRowProps({ style })} className={tableStyles.row}>
-            {row.cells.map((cell: Cell, index: number) => (
-              <TableCell
-                key={index}
-                field={data.fields[index]}
-                tableStyles={tableStyles}
-                cell={cell}
-                onCellClick={onCellClick}
-              />
-            ))}
-          </div>
-        );
-      },
-      [prepareRow, rows]
-    );
-
-    return (
-      <div {...getTableProps()} className={tableStyles.table}>
-        <CustomScrollbar hideVerticalTrack={true}>
-          <div style={{ width: `${totalColumnsWidth}px` }}>
-            {!noHeader && (
-              <div>
-                {headerGroups.map((headerGroup: HeaderGroup) => {
-                  return (
-                    <div className={tableStyles.thead} {...headerGroup.getHeaderGroupProps()} ref={ref}>
-                      {headerGroup.headers.map((column: Column, index: number) =>
-                        renderHeaderCell(column, tableStyles, data.fields[index])
-                      )}
-                    </div>
-                  );
-                })}
-              </div>
-            )}
-            <FixedSizeList
-              height={height - headerRowMeasurements.height}
-              itemCount={rows.length}
-              itemSize={tableStyles.rowHeight}
-              width={'100%'}
-              style={{ overflow: 'hidden auto' }}
-            >
-              {RenderRow}
-            </FixedSizeList>
-          </div>
-        </CustomScrollbar>
-      </div>
-    );
-  }
-);
+        case 'toggleSortBy':
+          if (props.onSortBy) {
+            // todo call callback and persist
+          }
+          break;
+      }
+
+      return newState;
+    },
+    [props.onColumnResize]
```

Comment:
```suggestion
    [props.onColumnResize, props.onSortBy]
```

Reply:
will fix when sort by persistence is added in this function